src/components/screens/AccountQRCodeScreen/AccountQRCodeScreen.tsx

```diff
@@ -49,23 +54,47 @@ const AccountQRCodeScreen: React.FC<AccountQRCodeScreenProps> = ({
     }
   }, [navigation, showNavigationAsCloseButton]);
 
-  useRightHeaderButton({
-    hidden: !showNavigationAsCloseButton,
-    icon: Icon.Scan,
-    onPress: () => {
-      const routes = navigation.getState()?.routes ?? [];
-      const scanRouteIndex = routes.findIndex(
-        (r) => r.name === ROOT_NAVIGATOR_ROUTES.SCAN_QR_CODE_SCREEN,
-      );
+  const getRightHeaderIcon = () =>
+    showNavigationAsCloseButton ? Icon.Scan : Icon.HelpCircle;
+
+  const handleWalletConnectNavigation = () => {
+    // WalletConnect flow: Navigate between QR Scan and QR Address screens
+    const routes = navigation.getState()?.routes ?? [];
+    const scanRouteIndex = routes.findIndex(
+      (r) => r.name === ROOT_NAVIGATOR_ROUTES.SCAN_QR_CODE_SCREEN,
+    );
 
-      // If the scan route is already in the stack, pop to it
-      // Otherwise, navigate to it
-      if (scanRouteIndex !== -1) {
-        navigation.popTo(ROOT_NAVIGATOR_ROUTES.SCAN_QR_CODE_SCREEN);
-      } else {
-        navigation.navigate(ROOT_NAVIGATOR_ROUTES.SCAN_QR_CODE_SCREEN);
+    // If the scan route is already in the stack, go back to it
+    // Otherwise, navigate to it
+    if (scanRouteIndex !== -1) {
+      // Calculate how many screens to go back
+      const currentIndex = routes.length - 1;
+      const stepsBack = currentIndex - scanRouteIndex;
+      for (let i = 0; i < stepsBack; i++) {
+        navigation.goBack();
```

Comment:
why not use `navigation.popTo(ROOT_NAVIGATOR_ROUTES.SCAN_QR_CODE_SCREEN);` in place of this loop?